Savanna/Model/Animal.cs

```diff
@@ -0,0 +1,196 @@
+
+        /// <summary>
+        /// Makes next move of animal.
+        /// </summary>
+        /// <param name="animal">Current animal.</param>
+        /// <param name="newAnimalsCoordinates">Free cells to possible move.</param>
+        public static void MakeNextRandomMove(Animal animal, List<AnimalNewCoordinates> newAnimalsCoordinates)
+        {
+            Random random = new();
+            AnimalNewCoordinates nextCoordinatesToMove = newAnimalsCoordinates[random.Next(0, newAnimalsCoordinates.Count)];
+            animal.XCoordinate = nextCoordinatesToMove.NewXCoordinate;
+            animal.YCoordinate = nextCoordinatesToMove.NewYCoordinate;
+        }
+
+        /// <summary>
+        /// Calculates possible free cells on board to move animal. 
+        /// </summary>
+        /// <param name="animals">List of animals.</param>
+        /// <param name="animal">Current animal.</param>
+        /// <returns>List of free cells for possible move for current animal.</returns>
+        protected static List<AnimalNewCoordinates> FreeCellsToMove(List<Animal> animals, Animal animal)
+        {
+            List<AnimalNewCoordinates> newAnimalsCoordinates = new();
+            AnimalNewCoordinates newCoordinates;
+
+            for (int newYCoordinate = animal.YCoordinate - 1; newYCoordinate <= animal.YCoordinate + 1; newYCoordinate++)
+            {
+                for (int newXCoordinate = animal.XCoordinate - 1; newXCoordinate <= animal.XCoordinate + 1; newXCoordinate++)
+                {
+                    if (!IsCellReserved(newYCoordinate, newXCoordinate, animals) && !Game.IsCellOnBoard(newYCoordinate, newXCoordinate))
+                    {
+                        newCoordinates = new AnimalNewCoordinates
+                        {
+                            NewYCoordinate = newYCoordinate,
+                            NewXCoordinate = newXCoordinate
+                        };
+
+                        newAnimalsCoordinates.Add(newCoordinates);
+                    }
+                }
+            }
+
+            newCoordinates = new AnimalNewCoordinates
+            {
+                NewYCoordinate = animal.YCoordinate,
+                NewXCoordinate = animal.XCoordinate
+            };
+
+            newAnimalsCoordinates.Add(newCoordinates);
+            return newAnimalsCoordinates;
+        }
+
+        /// <summary>
+        /// Checks if cell is reserved.
+        /// </summary>
+        /// <param name="newXCoordinate">New X position.</param>
+        /// <param name="newYCoordinate">New Y position.</param>
+        /// <param name="animals">List of animals.</param>
+        /// <returns>Is cell reserved.</returns>
+        private static bool IsCellReserved(int newXCoordinate, int newYCoordinate, List<Animal> animals)
+        {
+            for (int element = 0; element < animals.Count; element++)
+            {
+                Animal? animal = animals[element];
```

Comment:
You can try using LINQ here to simplify like animals.Exists(....)

Reply:
LINQ used for method.